readerwriterqueue.h

```diff
@@ -698,7 +698,7 @@ class AE_ALIGN(MOODYCAMEL_CACHE_LINE_SIZE) ReaderWriterQueue
 
 		// size must be a power of two (and greater than 0)
 		AE_NO_TSAN Block(size_t const& _size, char* _rawThis, char* _data)
-			: front(0), localTail(0), tail(0), localFront(0), next(nullptr), data(_data), sizeMask(_size - 1), rawThis(_rawThis)
+			: front(0UL), localTail(0), tail(0UL), localFront(0), next(nullptr), data(_data), sizeMask(_size - 1), rawThis(_rawThis)
```

Comment:
This shouldn't be necessary?

Reply:
Without `UL` we get:
```
g++  -std=c++11 -Wsign-conversion -Wpedantic -Wall -DNDEBUG -O3 -g unittests.cpp ../common/simplethread.cpp -o unittests -pthread
In file included from unittests.cpp:12:
In file included from ./../../readerwriterqueue.h:7:
./../../atomicops.h:247:61: warning: implicit conversion changes signedness: 'int' to 'unsigned long' [-Wsign-conversion]
        template<typename U> AE_NO_TSAN weak_atomic(U&& x) : value(std::forward<U>(x)) {  }
                                                             ~~~~~ ^~~~~~~~~~~~~~~~~~
./../../readerwriterqueue.h:701:6: note: in instantiation of function template specialization 'moodycamel::weak_atomic<unsigned long>::weak_atomic<int>' requested here
                        : front(0), localTail(0), tail(0), localFront(0), next(nullptr), data(_data), sizeMask(_size - 1), rawThis(_rawThis)
                          ^
./../../readerwriterqueue.h:726:32: note: in instantiation of member function 'moodycamel::ReaderWriterQueue<int, 512>::Block::Block' requested here
                return new (newBlockAligned) Block(capacity, newBlockRaw, newBlockData);
                                             ^
./../../readerwriterqueue.h:113:18: note: in instantiation of member function 'moodycamel::ReaderWriterQueue<int, 512>::make_block' requested here
                                auto block = make_block(largestBlockSize);
                                             ^
unittests.cpp:103:27: note: in instantiation of member function 'moodycamel::ReaderWriterQueue<int, 512>::ReaderWriterQueue' requested here
                        ReaderWriterQueue<int> q;
                                               ^
1 warning generated.
```

Reply:
Huh, OK, seems to be an artifact due to the argument forwarding of `weak_atomic<size_t>`.